### PR TITLE
Updated Flexvolume setup mechanisms for COS instance image.

### PR DESCRIPTION
**Pull request by verult**
- If REMOUNT_VOLUME_PLUGIN_DIR is set to true, VOLUME_PLUGIN_DIR is remounted with `exec` option during cluster startup. This allows any writable location to be used as the plugin directory.
- New HostPath added to controller-manager deployment to enable access to volume plugin directory.
- Improved how the default directory is passed to master and node setup.

**What this PR does / why we need it**: Part of the fix for #57353

/release-note-none
/sig storage
/assign @saad-ali @roberthbailey 
/cc @chakri-nelluri @wongma7

### CODE REVIEW

File: cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest [hunk 1]
```diff
@@ -137,6 +138,7 @@
         { "name": "srvkube",
         "mountPath": "{{srv_kube_path}}",
         "readOnly": true},
+        {{flexvolume_hostpath_mount}}
```
Comment (saad-ali) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: Does this apply to all kube-controller-manager deployments? Or just GCE? Also on GKE can users even deploy a Flex driver to a master machine?

Comment (verult) [reply cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: It applies to all deployments. No Flex drivers can't be deployed on GKE masters, so the HostPath will exist but won't be used.

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -402,3 +402,6 @@ ROTATE_CERTIFICATES="${ROTATE_CERTIFICATES:-}"
 # The number of services that are allowed to sync concurrently. Will be passed
 # into kube-controller-manager via `--concurrent-service-syncs`
 CONCURRENT_SERVICE_SYNCS="${CONCURRENT_SERVICE_SYNCS:-}"
+
+# Flexvolume plugin directory
+VOLUME_PLUGIN_DIR="${VOLUME_PLUGIN_DIR:-/etc/srv/kubernetes/kubelet-plugins/volume/exec}"
```
Comment (roberthbailey) [inline cluster/gce/config-default.sh]: Does this need to be set in config-test.sh as well?

Comment (roberthbailey) [reply cluster/gce/config-default.sh]: Does this need to be set conditionally based on the master / node OS type selected? Previously it was only set for gci, but this sets it unconditionally.

Comment (verult) [reply cluster/gce/config-default.sh]: This was already set in config-test.sh

Shoot, yes you are right. Reverting this back to the original.

**Review by wattli — APPROVED** [review]

**Comment by wattli** [comment]
Just let you know that, this will unblock the workload identity provisioning process of Istio Security. Thanks for facilitating this.

**Comment by verult** [comment]
Removed default REMOUNT_VOLUME_PLUGIN_DIR for this PR because /etc/srv/... doesn't require remounting. Will add it back in a subsequent PR for changing the default Flexvolume directory for COS.

**Comment by saad-ali** [comment]
This should be cherry-picked to 1.9 and 1.8

**Pull request merged**